pkg/lib/indexconfig/indexconfig.go

```diff
@@ -0,0 +1,124 @@
+	"github.com/operator-framework/operator-registry/pkg/image/execregistry"
+	"github.com/operator-framework/operator-registry/pkg/lib/certs"
+	registry "github.com/operator-framework/operator-registry/pkg/lib/registry"
+	regi "github.com/operator-framework/operator-registry/pkg/registry"
+
+	"github.com/sirupsen/logrus"
+)
+
+// ConfigUpdater is an implementation of ConfigAdder
+type ConfigUpdater struct {
+	Logger *logrus.Entry
+}
+
+// AddConfigRequest represents a request for a bundle to be added
+// to it's package config
+type AddConfigRequest struct {
+	Bundles       []string
+	ConfigFolder  string
+	ContainerTool containertools.ContainerTool
+	CaFile        string
+	SkipTLS       bool
+}
+
+// AddToConfig persists the bundle in it's package config
+func (c ConfigUpdater) AddToConfig(request AddConfigRequest) error {
+
+	var reg image.Registry
+	var rerr error
+	switch request.ContainerTool {
+	case containertools.NoneTool:
+		rootCAs, err := certs.RootCAs(request.CaFile)
+		if err != nil {
+			return fmt.Errorf("failed to get RootCAs: %v", err)
+		}
+		reg, rerr = containerdregistry.NewRegistry(containerdregistry.SkipTLS(request.SkipTLS), containerdregistry.WithRootCAs(rootCAs))
+	case containertools.PodmanTool:
+		fallthrough
+	case containertools.DockerTool:
+		reg, rerr = execregistry.NewRegistry(request.ContainerTool, c.Logger, containertools.SkipTLS(request.SkipTLS))
+	}
+	if rerr != nil {
+		return rerr
+	}
+	defer func() {
+		if err := reg.Destroy(); err != nil {
+			c.Logger.WithError(err).Warn("error destroying local cache")
+		}
+	}()
+
+	simpleRefs := make([]image.Reference, 0)
+	for _, ref := range request.Bundles {
+		simpleRefs = append(simpleRefs, image.SimpleReference(ref))
+	}
+
+	for _, ref := range simpleRefs {
+		to, from, cleanup, err := registry.UnpackImage(context.TODO(), reg, ref)
+		if err != nil {
+			return err
+		}
+		defer cleanup()
+
+		img, err := regi.NewImageInput(to, from)
+		if err := printConfig(img.Bundle, request.ConfigFolder); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func printConfig(bundle *regi.Bundle, configFolder string) error {
+
+	configFile := configFolder + "/" + bundle.Package + ".json"
+	configBundle, err := NewConfigBundle(bundle)
+	if err != nil {
+		return err
+	}
+	bundleJSON, err := json.MarshalIndent(configBundle, " ", "   ")
+	if err != nil {
+		return err
+	}
+	if err != nil {
+		return err
+	}
+	var f *os.File
+	defer f.Close()
+	if _, err = os.Stat(configFile); err == nil {
+		f, err = os.OpenFile(configFile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+		if err != nil {
+			return err
+		}
+	} else if os.IsNotExist(err) {
+		f, err = os.Create(configFile)
+		if err != nil {
+			return err
+		}
+		configPackage, err := NewConfigPackage(bundle)
+		if err != nil {
+			return err
+		}
+		pkgJSON, err := json.MarshalIndent(configPackage, " ", "   ")
+		if err != nil {
+			return err
+		}
+		f.Write(pkgJSON)
+		f.Write([]byte("\n"))
+	} else {
+		return err
+	}
```

Comment:
I'd recommend moving this to a helper function or moving around some of this logic to deflate any if/else if/else clause type patterns. It looks like you can just check if the err is nil, then open that file and write some bytes to it and return early, else check if the error is due to the file not existing and handle it from there.

Reply:
Looking at some of the other parts of the registry codebase, it seems like we have to perform this kind of check in a couple of different places. Probably not something we can generalize unless it's just checking for file existence and creating that file as needed, or handling errors as needed.